orca-applications/src/main/groovy/com/netflix/spinnaker/orca/applications/tasks/UpsertApplicationTask.groovy

```diff
@@ -40,25 +40,20 @@ class UpsertApplicationTask extends AbstractFront50Task {
      */
 
     def existingApplication = fetchApplication(application.name)
-    try {
-      if (existingApplication) {
-        outputs.previousState = existingApplication
-        log.info("Updating application (name: ${application.name})")
-        front50Service.update(application.name, application)
-      } else {
-        log.info("Creating application (name: ${application.name})")
-        front50Service.create(application)
-        if (application.permission?.permissions == null) {
-          application.setPermissions(Permissions.EMPTY)
-        }
+    if (existingApplication) {
+      outputs.previousState = existingApplication
+      log.info("Updating application (name: ${application.name})")
+      front50Service.update(application.name, application)
+    } else {
+      log.info("Creating application (name: ${application.name})")
+      front50Service.create(application)
+      if (application.permission?.permissions == null) {
+        application.setPermissions(Permissions.EMPTY)
       }
+    }
 
-      if (application.permission?.permissions != null) {
-        front50Service.updatePermission(application.name, application.permission)
-      }
-    } catch (RetrofitError re) {
-      log.error("Could not create or update application permission", re)
-      return new TaskResult(ExecutionStatus.TERMINAL, [:], outputs)
+    if (application.permission?.permissions != null) {
+      front50Service.updatePermission(application.name, application.permission)
```

Comment:
If one of the front50Service calls fail, does the TaskResult return a `TERMINAL` state? I think that's why I wrapped the rest of the front50 calls in the try in the first place. Not sure what the default handling will do here.

Reply:
Yup, the task/execution is being flagged `TERMINAL`.